src/Hng.Application/Services/UserService.cs

```diff
@@ -37,5 +53,78 @@ public async Task<IEnumerable<UserDto>> GetAllUsersAsync()
             return _mapper.Map<IEnumerable<UserDto>>(users);
             
         }
+
+
+
+
+
+
+
+
+
+        public async Task<(bool IsSuccess, string ErrorMessage)> IsEmailUniqueAsync(string email)
+        {
+            var existingUser = await _context.Users.FirstOrDefaultAsync(u => u.Email == email);
```

Comment:
Don't use the data context directly. Use the `UserRepository`. In fact this whole method should be in the user repository as it directly interacts with the database.